### PR TITLE
Allow users to disable metrics via environment variable: 'LS_METRICS_ENABLED=false'

### PR DESCRIPTION
**Pull request by malafeev**

### CODE REVIEW

File: common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java [hunk 1]
```diff
@@ -259,7 +259,8 @@ private void maybeStartReporting() {
         reportingThread.setDaemon(true);
         reportingThread.start();
 
-        if (!disableMetricsReporting && safeMetrics != null) {
+        if (!disableMetricsReporting && safeMetrics != null
```
Comment (carlosalberto) [inline common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java:262]: My impression is that the `disableMetricsReporting` attribute is the one that should be directly set from `Options`, so we should be able to override it there.

```java
  // Options
  private boolean disableMetricsReporting = getEnvMetricsEnabled();

  static boolean getEnvMetricsEnabled() {
    String envMetricsEnabled = System.getenv(...);
    if (envMetricsEnabled != null) {
      // Return true/false if parsed properly.
      // Else, log the invalid value, and...
    }
    // ... Return DEFAULT_DISABLE_METRICS; // false
  }
```

Comment (malafeev) [reply common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java]: updated

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Thanks for the update! maybe we can add a test case for `Options`? Either in this PR or in a follow up.

**Comment by malafeev** [comment]
@carlosalberto I added tests

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**